### PR TITLE
Validation tests

### PR DESCRIPTION
**Pull request by RolandMacDoland**
This PR addresses the addition of validation tests under the form of a round trip QIR -> pytket circuit -> QIR and (partially) validating the equivalence of outputs up to irrelevant reordering of function calls.

It also fixes bugs around the rebase of parametrised gates and better handling of data generation for runtime function calls.

### CODE REVIEW

File: tests/generator_test.py
Comment (cqc-alec) [inline tests/generator_test.py]: Is this something you will be returning to? Maybe make a Jira task?

**Review by cqc-alec — CHANGES_REQUESTED** [review]
Please remove binary wheel!

**Comment by RolandMacDoland** [comment]
> Please remove binary wheel!

Yes sorry. It keeps coming back as part of the merge from other branches.

File: tests/generator_test.py
Comment (RolandMacDoland) [reply tests/generator_test.py]: Well, hopefully yes but it would certainly need to think about the non-simplicity of `CircBox` in `pytket`.

Comment (Roland-djee) [reply tests/generator_test.py:443]: In the Jira board [here](https://cqc.atlassian.net/browse/TKET-2662?atlOrigin=eyJpIjoiNTk2ZDI1OGJlMGY4NDdmM2JjYTIxMTdhOTc5NjYzOTMiLCJwIjoiaiJ9)

Comment (RolandMacDoland) [reply tests/generator_test.py]: In the Jira board now.

Comment (cqc-alec) [reply tests/generator_test.py]: Thanks. Could you describe what this means in terms of pytket-pyqir functionality?

- What pytket-pyqir functionality is missing because of the limitations of the pytket `CircBox` API?
- What change in pytket is needed to unblock this?

Comment (RolandMacDoland) [reply tests/generator_test.py]: Sure, I can try my best to describe that.

File: pytket_qir/generator.py
Comment (cqc-alec) [inline pytket_qir/generator.py]: Do we need this branchimg? I think it should just work if `params == []`.

File: pytket_qir/parser.py
Comment (cqc-alec) [inline pytket_qir/parser.py]: Can this happen?

**Review by cqc-alec — CHANGES_REQUESTED** [review]

File: pytket_qir/generator.py
Comment (RolandMacDoland) [reply pytket_qir/generator.py]: Indeed. Thanks.

File: pytket_qir/parser.py
Comment (RolandMacDoland) [reply pytket_qir/parser.py]: Nope that's right. Removed.

**Review by cqc-alec — COMMENTED** [review]
Since we are disabling tests here, I'd like to make sure the issue is properly tracked before merging. Could you please make a new issue on this repo for the flattening/tracking issue, and reference it from the skipped test code? Then I'm happy for this to be merged. Thanks.

**Comment by RolandMacDoland** [comment]
> Since we are disabling tests here, I'd like to make sure the issue is properly tracked before merging. Could you please make a new issue on this repo for the flattening/tracking issue, and reference it from the skipped test code? Then I'm happy for this to be merged. Thanks.

OK sure. Will do now.

**Pull request merged**
